packages/tui/internal/components/chat/message.go

```diff
@@ -22,6 +24,38 @@ import (
 	"golang.org/x/text/language"
 )
 
+// shouldUse24HourFormat determines if 24-hour time format should be used
+// based on config and system detection
+func shouldUse24HourFormat(config *opencode.Config) bool {
+	switch config.TimeFormat {
+	case "24h":
+		return true
+	case "12h":
+		return false
+	default:
+		// Default to detect behavior
+		if runtime.GOOS == "darwin" {
+			return detectMacOS24HourFormat()
+		}
+		return false // fallback to 12h on other platforms
```

Comment:
i like the UX of it auto-determining on mac os, but with the way bubbletea works this code is going to run a lot; we should either 1) drop the auto-detect and just default to 12h unless you configure or 2) store the detected format in `app.State` and use it if there's no configured value.

Reply:
Done, I think this is what you're more expecting. I'd typically use a sync.OnceValue or something here, but I don't think there's anything that could happen as a race condition, so just used a *bool on the app.State.